gemac/management.py

```diff
@@ -0,0 +1,155 @@
+from myhdl import block, Signal, intbv, always_seq, always_comb
+from myhdl._ShadowSignal import TristateSignal
+
+
+rx0, rx1, tx, flow, managementreg, ucast0, \
+    ucast1, addrtable0, addrtable1, addrfiltermode = range(10)
+hostclk_count = 0
+
+
+class mdioData:
+    def __init__(self, opcode, hostaddr, wrdata):
+        self.pre = intbv(0xFFFFFFFF)[32:]
+        self.st = intbv(0b01)[2:]
+        self.op = opcode[2:]
+        self.phyadd = hostaddr[10:5]
+        self.regadd = hostaddr[5:]
+        self.wrdata = wrdata[16:]
+
+
+@block
+def management(host_interface, mdio_interface):
+
+    """
+    Management Data - Input Output Interface.
```

Comment:
Yes, I shall work on documentation as soon I feel the completeness of the block.
Well this module is mainly for making the configuration editable by the host during the application and also to covert host transactions to MII MDIO transactions as you mentioned.
So if you want host-less application(is it a compulsion?) this module should not be present at all.
I am thinking of developing this and give the designer choice whether he wants the gemac core with or without host using parameters using [conditional instantiation](http://docs.myhdl.org/en/stable/manual/structure.html#conditional-instantiation).
Will embedded processor will be required for anything other than host interface considering the direction i am going now?